### PR TITLE
Jenkins: print systemd logs

### PR DESCRIPTION
**Pull request by ggreer**
- Add forensic script to ssh to machine & print debugging info
- Add cluster-foreach script to run forensics script on each machine in the ASG.
- Get jenkins & smoke tests to run these scripts automatically (even on failure).

### CODE REVIEW

**Comment by ggreer** [comment]
Closed #1168 & opened a new PR so that it will run in another region. I'm tired of waiting for cleanup scripts to run.

File: tests/smoke/aws/cluster-foreach.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+set -x
+set -e
+
```
Comment (squat) [inline tests/smoke/aws/cluster-foreach.sh]: Could you `set -o pipefail`?

Comment (kans) [inline tests/smoke/aws/cluster-foreach.sh]: I don't think we want x or pipefail here - this is just for debugging.

File: tests/smoke/forensics.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+set -x
+
+echo -n "hostname: "
```
Comment (squat) [inline tests/smoke/forensics.sh]: These would all be more succinct with `printf "hostname %s\n" "$(hostname)"`

File: tests/smoke/aws/cluster-foreach.sh (hunk 1 above)
Comment (squat) [inline tests/smoke/aws/cluster-foreach.sh]: Ah ok fair enuff. No pipefail, e; yes x.

File: Jenkinsfile [hunk 3]
```diff
@@ -79,7 +80,7 @@ pipeline {
           "SmokeTest TerraForm: AWS": {
             node('worker && ec2') {
               withCredentials(creds) {
-                withDockerContainer(builder_image) {
+                withDockerContainer(args: '-v /etc/passwd:/etc/passwd', image: builder_image) {
```
Comment (Quentin-M) [inline Jenkinsfile]: Is there something else we can do..? Like useradd on the executor/in the image, changing the UID/GID mapping..? Terraform could be made to do anything with that.

**Comment by ggreer** [comment]
`/etc/passwd` doesn't contain password hashes. It's just a list of usernames, user ids, group ids, and default shells. Also, I changed jenkins to mount it read-only.

**Comment by ggreer** [comment]
Anyone want to chime in here? I'm pretty sure it's fine to mount `/etc/passwd` as read-only.

**Comment by kans** [comment]
It would probably be a good idea to add `kubelet` to the list - [see also](https://github.com/coreos/tectonic-installer/blob/master/Documentation/troubleshooting/master-nodes.md)

**Comment by ggreer** [comment]
Probably a good idea, but this PR has been open long enough that I just want to get it merged. We can improve on the reporting after that.

**Pull request merged**